lib/OpenQA/Schema/Result/ScheduledProducts.pm

```diff
@@ -753,4 +727,122 @@ sub _create_download_lists {
     }
 }
 
+sub _schedule_from_yaml ($self, $args, $skip_chained_deps, @load_yaml_args) {
+    my $data = eval { load_yaml(@load_yaml_args) };
+    if (my $error = $@) { return {error_message => "Unable to load YAML: $error"} }
+    my $app = OpenQA::App->singleton;
+    my $validation_errors = $app->validate_yaml($data, 'JobScenarios-01.yaml', $app->log->level eq 'debug');
+    return {error_message => "YAML validation failed:\n" . join("\n", @$validation_errors)} if @$validation_errors;
+
+    my $products = $data->{products};
+    my $machines = $data->{machines};
+    my $job_templates = $data->{job_templates};
+    my $error_msg;
+    my %wanted;
+    my @job_templates;
+    foreach my $key (keys %$job_templates) {
```

Comment:
My usual suggestion to iterate over sorted keys to ensure reproducability:
```suggestion
    foreach my $key (sort keys %$job_templates) {
```

Reply:
The "normal" code (using the DB) doesn't sort job templates. I'm not sure whether it is worth sorting this data (as it also has a cost).

Reply:
The cost of sort can be ignored here. This function will be called rarely enough and the number of items will be small enough.
But I don't insist...